gsy_framework/validators/profile_validator.py

```diff
@@ -0,0 +1,44 @@
+from collections import OrderedDict
+from datetime import timedelta
+from typing import Dict
+
+from pendulum import DateTime, duration
+
+from gsy_framework.utils import convert_str_to_pendulum_in_dict
+
+
+class ProfileValidator:
+    """Validate profiles"""
+
+    @classmethod
+    def validate(cls, profile: Dict[str, float], start_time: DateTime, end_time: DateTime,
+                 slot_length: timedelta):
+        """Validate if profile corresponds to the start_time, end_time and slot_length setting."""
+        assert len(profile) > 0, "profile is empty"
+        ordered_profile = OrderedDict(convert_str_to_pendulum_in_dict(profile))
+        cls._validate_start_end_date(ordered_profile, start_time, end_time, slot_length)
+        cls._validate_slot_length(ordered_profile, slot_length)
+
+    @staticmethod
+    def _validate_start_end_date(
+            profile: Dict[DateTime, float], start_time: DateTime, end_time: DateTime,
```

Comment:
This should be OrderedDict, trivial otherwise however it helps understanding the calculation of profile_start_time and profile_end_time